### PR TITLE
feat: bookmark editor

### PR DESCRIPTION
**Pull request by nesquikm**
## Description

This PR includes full functional bookmark editor.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/app/service/storage_service/browser_bookmarks_storage_service.dart [hunk 1]
```diff
@@ -70,12 +70,26 @@ class BrowserBookmarksStorageService extends AbstractStorageService {
...
-  Future<void> clearBrowserBookmarks() async {
+  Future<void> clearBrowserBookmarks({
+    bool needUndo = true,
+  }) async {
     await _storage.delete(
       _browserBookmarksKey,
       domain: _browserBookmarksDomain,
     );
 
+    final savedbrowserBookmarks = browserBookmarks;
```
Comment (Alex-A4) [inline lib/app/service/storage_service/browser_bookmarks_storage_service.dart:81]: I think, you should save this variable before doing `delete` action, isn't it?

Comment (Alex-A4) [reply lib/app/service/storage_service/browser_bookmarks_storage_service.dart]: And it looks like we should add same logic for deleting single bookmark

Comment (nesquikm) [reply lib/app/service/storage_service/browser_bookmarks_storage_service.dart]: Yeah, thank you very much! Fixed.

File: lib/data/models/browser_bookmark_item.dart [hunk 2]
```diff
@@ -28,6 +28,6 @@ class BrowserBookmarkItem with _$BrowserBookmarkItem {
         title: title,
         url: url,
         faviconUrl: faviconUrl,
-        sortingOrder: DateTime.now().millisecondsSinceEpoch,
+        sortingOrder: DateTime.now().millisecondsSinceEpoch.toDouble(),
```
Comment (Alex-A4) [inline lib/data/models/browser_bookmark_item.dart:31]: What the problem with int?

File: lib/app/service/storage_service/browser_bookmarks_storage_service.dart (hunk 1 above)
Comment (nesquikm) [reply lib/app/service/storage_service/browser_bookmarks_storage_service.dart]: When deleting single bookmark we have `item` object that we can restore when undo'ing.

**Review by Alex-A4 — APPROVED** [review]

File: lib/data/models/browser_bookmark_item.dart (hunk 2 above)
Comment (nesquikm) [reply lib/data/models/browser_bookmark_item.dart:31]: It's just a cheap sort pattern: we can put item between two other by calculating sorting order index as `(next + prev) / 2 + prev` without rewriting other items.

**Pull request merged**